### DISCUSSION TITLE
Run custom function

### DISCUSSION THREAD

**Discussion by thecookieswhoplays** [pocketbase/pocketbase#3291] [Q&A]
I need to make some kind of funtcion which uses the discord api to make a user join a server however discord api can not be used from a browser. I need to make the function only available to admins so it doesn't get abused. Also another question I had is if it's possible to make a table only accessible to people who are marked as admin

**Comment by ganigeorgiev** [comment]
For executing custom server-side code you can check https://pocketbase.io/docs/use-as-framework/ (_and more specifically see the "Routing" section depending on your language of choice_).
Alternatively, if you already use a SSR framework like SvelteKit, Netxt.js, etc. you can define your custom handler in a custom node server action and for example use the JS SDK as admin PocketBase account.

---

> Also another question I had is if it's possible to make a table only accessible to people who are marked as admin

It depends what do you mean by "marked as admin". If you have a "role" field for example you can set an API rule like: `@request.auth.role = "admin"`.

Note that in order to prevent users to change their own role you can use the following API rules for the `users` collection: `@request.data.role:isset = false` or `@request.data.role = "user"` (_these rules practically allow only PocketBase admins to change the user record role_).
If you haven't already, you can also check [API rules and filters](https://pocketbase.io/docs/api-rules-and-filters/) for more details on the API rules and filter syntax.
